tests/models/groupvit/test_modeling_groupvit.py

```diff
@@ -161,6 +163,18 @@ def test_config(self):
     def test_inputs_embeds(self):
         pass
 
+    @is_pt_tf_cross_test
+    def test_pt_tf_model_equivalence(self):
+        import tensorflow as tf
+
+        seed = 338
+        random.seed(seed)
+        np.random.seed(seed)
+        torch.manual_seed(seed)
+        torch.cuda.manual_seed_all(seed)
+        tf.random.set_seed(seed)
```

Comment:
Is this because the test fails for most seeds?

Reply:
Hi, @gante. Yes. The test issue is:

`GroupViT` computes some indices using argmax (on `y_soft` in the code below), uses them as one-hot encoding for further computation.

The problem: while PT/TF have very small difference in `y_soft`  (~ `1e-9`), the argmax `could be` totally different, if there are at least 2 indices with `almost identical values`. This leads to very large difference in the outputs.
```
    y_soft = logits.softmax(dim) --> this could have very small difference
    # Straight through.
    index = y_soft.max(dim, keepdim=True)[1] --> this could have different indices
    y_hard = torch.zeros_like(logits, memory_format=torch.legacy_contiguous_format).scatter_(dim, index, 1.0) --> this could have large difference (i.e. 1.0)
```

We need specific seeds to avoid `almost identical values` happening in `y_soft`. This is not very desirable, but that's the only solution I could come up with at this moment.

Reply:
I tell @ariG23498 to add comments about this in the code :-)

Reply:
For future reference: the PT/TF equivalence test has `tf_model = transformers.load_pytorch_model_in_tf2_model(tf_model, pt_model, tf_inputs=tf_inputs_dict)`, so the TF model weighs is determined by the PT weights.

So we only need to fix the inputs to avoid the problematic mentioned above. The tensors are generated using Python `random.Random()`. So we only need to set Python and torch seeds.

Reply:
Thanks for the additional information @ydshieh!